pkg/reconciler/pipelinerun/cancel_test.go

```diff
@@ -104,46 +110,189 @@ func TestCancelPipelineRun(t *testing.T) {
 				Status: v1beta1.PipelineRunSpecStatusCancelledDeprecated,
 			},
 		},
+	}, {
+		name: "child-references",
+		pipelineRun: &v1beta1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-pipeline-run-cancelled"},
+			Spec: v1beta1.PipelineRunSpec{
+				Status: v1beta1.PipelineRunSpecStatusCancelled,
+			},
+			Status: v1beta1.PipelineRunStatus{PipelineRunStatusFields: v1beta1.PipelineRunStatusFields{
+				ChildReferences: []v1beta1.ChildStatusReference{
+					{
+						TypeMeta:         runtime.TypeMeta{Kind: "TaskRun"},
+						Name:             "t1",
+						PipelineTaskName: "task-1",
+					},
+					{
+						TypeMeta:         runtime.TypeMeta{Kind: "TaskRun"},
+						Name:             "t2",
+						PipelineTaskName: "task-2",
+					},
+					{
+						TypeMeta:         runtime.TypeMeta{Kind: "Run"},
+						Name:             "r1",
+						PipelineTaskName: "run-1",
+					},
+					{
+						TypeMeta:         runtime.TypeMeta{Kind: "Run"},
+						Name:             "r2",
+						PipelineTaskName: "run-2",
+					},
+				},
+				// If ChildReferences is not empty, TaskRuns and Runs should be ignored.
```

Comment:
This test may not strictly be necessary - it's a scenario that I don't _think_ we can ever actually encounter, since we have solid testing around how `pr.Status.TaskRuns`/`pr.Status.Runs` and `pr.Status.ChildReferences` get populated (depending on the `embedded-status` value) elsewhere. But I decided that I wanted to have testing of the fact that the cancel behavior ignores `pr.Status.TaskRuns` and `pr.Status.Runs` if `pr.Status.ChildReferences` isn't empty, just to be safe. =)

Reply:
I also don't think we can ever encounter it, but if it does occur it would be better to alert us via an error rather than silently succeeding. I agree this case should be tested.

Reply:
Ok, I literally can't find a way we ever could encounter this situation, and if we did, it would be cleaned up on reconcile anyway. I feel like my thinking was "what happens with a running PR that somehow gets borked when the feature flag value changes from `full` to `minimal`?" But the situation I put in this test wouldn't result from that case, so the "ignore `.status.taskRuns` and `.status.runs` if `.status.childReferences` is populated, then make sure that we don't do anything to whatever's in `.status.taskRuns` or `.status.runs`" scenario tested here is silly. I'll clean up the test.